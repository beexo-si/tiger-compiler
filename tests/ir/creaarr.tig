/*TIGER_ARGS=-arbol -ir*/
let
    type A = array of int
    var a:=A[100] of 51
in
    15
    /*a[20]*/
end
