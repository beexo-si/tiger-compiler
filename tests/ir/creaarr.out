LetExp{decs=[TypeDec([{name=A,ty=ArrayTy(int)}])
             VarDec{name=a,escape=false,typ=NONE,
              init=ArrayExp{test=A,size=100,init=51}}],body=15}
bien!
;;-PROC-L0__tigermain_0--:
SEQ(
 MOVE(
  TEMP T2,
  TEMP rbx),
 SEQ(
  MOVE(
   TEMP T3,
   TEMP r12),
  SEQ(
   MOVE(
    TEMP T4,
    TEMP r13),
   SEQ(
    MOVE(
     TEMP T5,
     TEMP r14),
    SEQ(
     MOVE(
      TEMP T6,
      TEMP r15),
     SEQ(
      MOVE(
       MEM(
        BINOP(PLUS,
         TEMP rbp,
         CONST -8)),
       TEMP rdi),
      SEQ(
       MOVE(
        TEMP rax,
        ESEQ(
         MOVE(
          TEMP T1,
          ESEQ(
           SEQ(
            EXP(
             CALL(
              NAME _initArray,
               CONST 100,
               CONST 51)),
            MOVE(
             TEMP T0,
             TEMP rax)),
           TEMP T0)),
         CONST 15)),
       SEQ(
        MOVE(
         TEMP rbx,
         TEMP T2),
        SEQ(
         MOVE(
          TEMP r12,
          TEMP T3),
         SEQ(
          MOVE(
           TEMP r13,
           TEMP T4),
          SEQ(
           MOVE(
            TEMP r14,
            TEMP T5),
           MOVE(
            TEMP r15,
            TEMP T6))))))))))))
;;-END-PROC--:
yes!!
Return code: 15
